internal/bundle/utils.go

```diff
@@ -88,26 +113,61 @@ func LoadWasmResolversFromStore(ctx context.Context, store storage.Store, txn st
 }
 
 // LoadBundleFromDisk loads a previously persisted activated bundle from disk
-func LoadBundleFromDisk(path, name string, bvc *bundle.VerificationConfig) (*bundle.Bundle, error) {
-	bundlePath := filepath.Join(path, name, "bundle.tar.gz")
+func LoadBundleFromDisk(path string, opts *LoadOptions) (*bundle.Bundle, error) {
+	// if a bundle package exists, use that as it might contain the bundle etag which
+	// is not stored in the legacy bundle file. This can help avoid unnecessary bundle
+	// downloads.
+	bundlePackagePath := filepath.Join(path, PackageFileName)
+	if _, err := os.Stat(bundlePackagePath); err == nil {
+		f, err := os.Open(filepath.Join(bundlePackagePath))
+		if err != nil {
+			return nil, fmt.Errorf("failed to open bundle package file: %w", err)
+		}
+
+		var bundlePackage bundlePackage
+		err = json.NewDecoder(f).Decode(&bundlePackage)
+		if err != nil {
+			return nil, fmt.Errorf("failed to decode bundle package file: %w", err)
+		}
+
+		r := bundle.NewReader(bytes.NewReader(bundlePackage.Bundle))
+		if opts.VerificationConfig != nil {
+			r = r.WithBundleVerificationConfig(opts.VerificationConfig)
```

Comment:
Lets check `opts` is not `nil`

Reply:
Thanks! see: 5cd8f2e8d74cc93d27a1273184987657b9295cb0